version-upgrade-operator/README.md

```diff
@@ -0,0 +1,629 @@
+---
+creation_date: 2025-05-29
+issues:
+  - https://github.com/giantswarm/roadmap/issues/4001
+last_review_date: 2025-05-29
+owners:
+  - https://github.com/orgs/giantswarm/teams/team-honeybadger
+state: review
+summary:
+  This RFC presents a solution that will allow our customers and us to configure automated app upgrades,
+  including cluster apps, according to a pre-defined policy, that includes time schedule and version
+  restrictions.
+---
+
+# Automatic app upgrades
+
+The goal is to be able to schedule automated app upgrades (Helm chart version upgrades) for deployments
+present on Management Clusters. The proposed solution has to support cluster deployment apps.
+
+## Assumptions and requirements
+
+- has to work without GitOps, but must be able to integrate with it
+- has to support schedules
+- works on MC, but targets any resources (so including App CRs and HelmReleases targeting remote WCs)
+
+## Problems
+
+Won't work with the current cluster Release process, where chart version is not really the version deployed.
```

Comment:
@alex-dabija to drop it from my brain, i reckon cluster upgrades are the biggest user here, so we need to make sure we get along with the chart version issue